service/src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceServiceTest.java

```diff
@@ -1348,41 +1348,54 @@ void updateBqDatasetWithInvalidExpirationTimes() throws Exception {
-    final ApiGcpBigQueryDatasetUpdateParameters updateParameters2 =
-        new ApiGcpBigQueryDatasetUpdateParameters()
-            .defaultTableLifetime(3600L)
-            .defaultPartitionLifetime(-2L);
-    assertThrows(
-        BadRequestException.class,
-        () ->
-            controlledResourceService.updateBqDataset(
-                resource, updateParameters2, null, null, userAccessUtils.defaultUserAuthRequest()));
-
-    // check the expiration times stored on the cloud are still undefined, because the update above
-    // failed
-    validateBigQueryDatasetCloudMetadata(
-        projectId, createdDataset.getDatasetName(), location, null, null);
+    try {
+      // make an update request to set the table expiration time to an invalid value (<3600)
+      final ApiGcpBigQueryDatasetUpdateParameters updateParameters =
+          new ApiGcpBigQueryDatasetUpdateParameters()
+              .defaultTableLifetime(3000L)
+              .defaultPartitionLifetime(3601L);
+      assertThrows(
+          BadRequestException.class,
+          () ->
+              controlledResourceService.updateBqDataset(
+                  resource,
+                  updateParameters,
+                  null,
+                  null,
+                  userAccessUtils.defaultUserAuthRequest()));
+
+      // check the expiration times stored on the cloud are still undefined, because the update
+      // above
+      // failed
+      validateBigQueryDatasetCloudMetadata(
+          projectId, createdDataset.getDatasetName(), location, null, null);
+
+      // make another update request to set the partition expiration time to an invalid value (<0)
+      final ApiGcpBigQueryDatasetUpdateParameters updateParameters2 =
+          new ApiGcpBigQueryDatasetUpdateParameters()
+              .defaultTableLifetime(3600L)
+              .defaultPartitionLifetime(-2L);
+      assertThrows(
+          BadRequestException.class,
+          () ->
+              controlledResourceService.updateBqDataset(
+                  resource,
+                  updateParameters2,
+                  null,
+                  null,
+                  userAccessUtils.defaultUserAuthRequest()));
+
+      // check the expiration times stored on the cloud are still undefined, because the update
+      // above
+      // failed
+      validateBigQueryDatasetCloudMetadata(
+          projectId, createdDataset.getDatasetName(), location, null, null);
 
-    // Remove dataset to not conflict with other test that checks for empty lifetime
-    controlledResourceService.deleteControlledResourceSync(
-        workspaceId, resource.getResourceId(), userAccessUtils.defaultUserAuthRequest());
+    } finally {
+      // Remove dataset to not conflict with other test that checks for empty lifetime
+      controlledResourceService.deleteControlledResourceSync(
+          workspaceId, resource.getResourceId(), userAccessUtils.defaultUserAuthRequest());
+    }
   }
```

Comment:
Surrounding this test with a `try... finally` clause so the BQ dataset is always removed.